virtualpdu/core.py

```diff
@@ -42,13 +43,25 @@ def pdu_outlet_state_changed(self, pdu, outlet, state):
         except KeyError:
             return
 
+        self._async_change(device, state)
+
+    def _async_change(self, device, state):
         if state == POWER_ON:
-            self.driver.power_on(device)
+            def switch_power():
+                self.driver.power_on(device)
         elif state == POWER_OFF:
-            self.driver.power_off(device)
+            def switch_power():
+                self.driver.power_off(device)
         elif state == REBOOT:
-            self.driver.power_off(device)
-            self.driver.power_on(device)
+            def switch_power():
+                self.driver.power_off(device)
+                self.driver.power_on(device)
+        else:
+            self.logger.error("Unknown power state: {}".format(state))
+            return
+
+        with ThreadPoolExecutor(max_workers=1) as executor:
```

Comment:
I would prefer seeing asyncio here As we are [Using asynccore] for the transport. A mixup of threadpool and asynccore doesn't seems a good Idea. However I would refactor this as an experiment and not related to this review.

[Using asynccore]: https://github.com/internap/virtualpdu/blob/master/virtualpdu/pdu/pysnmp_handler.py#L127